dns_reverse_proxy.go

```diff
@@ -100,8 +123,15 @@ func route(w dns.ResponseWriter, req *dns.Msg) {
 		dns.HandleFailed(w, req)
 		return
 	}
-	for name, addr := range routes {
+	for name, addrs := range routes {
 		if strings.HasSuffix(req.Question[0].Name, name) {
+			var addr string
```

Comment:
How about avoiding the `else` by using the default value for the case where there's only one element:
```
addr := addrs[0]
if n := len(addrs); n > 1 {
  addr = addrs[rand.Intn(n)]
}
```

Reply:
personally, i dont like multiple sets (performance reason), but this isnt a performance critic place, so right, codestyle is better here :)